### PR TITLE
add case for memory allocation modification

### PR DESCRIPTION
**Pull request by nanli1**
VIRT-297052: Memory allocation config modification
Signed-off-by: nanli <nanli@redhat.com>

### CODE REVIEW

**Comment by nanli1** [comment]
```
 /usr/bin/python /usr/local/bin/avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 memory.allocation.config.positive_test
 (1/2) type_specific.io-github-autotest-libvirt.memory.allocation.config.positive_test.without_numa: PASS (47.22 s)
 (2/2) type_specific.io-github-autotest-libvirt.memory.allocation.config.positive_test.with_numa: PASS (47.13 s)

```

**Comment by nanli1** [comment]
modular_redesign

File: libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        edit_cmd = r"::%s/{0}/{1}".format(mem_xml % mem_value, mem_xml % max_mem)
+        status = libvirt.exec_virsh_edit(vm_name, [edit_cmd])
+        if not status:
+            test.fail('Edit guest failed')
+        check_value = ''
+        if mem_config == "without_numa":
+            check_value = max_mem
+        elif mem_config == "with_numa":
+            check_value = mem_value
+        libvirt_vmxml.check_guest_xml(vm_name, mem_xml % check_value)
```
Comment (chunfuwen) [inline libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py:65]: it need confirm with @dzhengfy  whether libvirt_vmxml.check_guest_xml is now deprecated,

File: libvirt/tests/cfg/mem/memory_allocation/memory_allocation_config_modification.cfg [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    mem_xml = "<memory unit='${mem_unit}'>%s"
+    current_mem_xml = "<currentMemory unit='${current_mem_unit}'>%s"
+    active_maxmem_error = "cannot resize the maximum memory on an active domain"
+    variants case:
+        - positive_test:
+            variants mem_config:
+                - without_numa:
+                    vm_attrs = {'memory_unit':'${mem_unit}','memory':${mem_value},'current_mem':${current_mem},'current_mem_unit':"${current_mem_unit}"}
+                - with_numa:
+                    vm_attrs = {'max_mem_rt': ${max_mem}, 'max_mem_rt_slots': ${max_mem_slots}, 'max_mem_rt_unit': "${max_mem_unit}",'memory_unit':"${mem_unit}",'memory':${mem_value},'current_mem':${current_mem},'current_mem_unit':'${current_mem_unit}','vcpu': 4,'cpu': {'numa_cell': [{'id': '0', 'cpus': '0-1', 'memory': '${numa_mem}', 'unit': 'KiB'},{'id':'1','cpus': '2-3','memory':'${numa_mem}','unit':'KiB'}]}}
```
Comment (dzhengfy) [inline libvirt/tests/cfg/mem/memory_allocation/memory_allocation_config_modification.cfg:21]: To improve the readability, please divide it into several parameters.

File: libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
+from virttest import virsh
```
Comment (dzhengfy) [inline libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py:1]: Could you add the template of copyright?

File: libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        5.Start the guest then change memory to maxMemory by virsh setmaxmem
+        """
+        test.log.info("TEST_STEP1: Increase currentMemory value same with "
+                      "memory by virsh edit")
+        edit_cmd = r":%s/{0}/{1}".format(current_mem_xml % current_mem,
+                                         current_mem_xml % mem_value)
+        status = libvirt.exec_virsh_edit(vm_name, [edit_cmd])
+        if not status:
+            test.fail('Edit guest failed')
+        libvirt_vmxml.check_guest_xml(vm_name, current_mem_xml % mem_value)
```
Comment (dzhengfy) [inline libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py:43]: Consider to reuse 
- https://github.com/avocado-framework/avocado-vt/pull/3675

File: libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+        test.log.info("TEST_STEP2: Increase currentMemory value bigger"
+                      " than memory by virsh edit")
+        edit_cmd = r":%s/{0}/{1}".format(current_mem_xml % current_mem,
+                                         current_mem_xml % bigger_mem)
+        status = libvirt.exec_virsh_edit(vm_name, [edit_cmd])
+        if not status:
+            test.fail('Edit guest failed')
+        libvirt_vmxml.check_guest_xml(vm_name, current_mem_xml % mem_value)
```
Comment (dzhengfy) [inline libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py:52]: same as above

File: libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py (hunk 1 above)
Comment (dzhengfy) [reply libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py:65]: reuse 
- https://github.com/avocado-framework/avocado-vt/pull/3675

File: libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+        test.log.info("TEST_STEP4:Increase memory value to maxMemory by "
+                      "cmd virsh setmaxmem")
+        ret = virsh.setmaxmem(vm_name, max_mem)
+        if mem_config == "without_numa":
+            libvirt.check_exit_status(ret)
+            libvirt_vmxml.check_guest_xml(vm_name, mem_xml % max_mem)
+        elif mem_config == "with_numa":
+            libvirt.check_exit_status(ret, maxmem_error)
+            libvirt_vmxml.check_guest_xml(vm_name, mem_xml % mem_value)
```
Comment (dzhengfy) [inline libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py:75]: same as above

File: libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        elif mem_config == "with_numa":
+            check_value = mem_value
+        libvirt_vmxml.check_guest_xml(vm_name, mem_xml % check_value)
+
+        test.log.info("TEST_STEP4:Increase memory value to maxMemory by "
+                      "cmd virsh setmaxmem")
+        ret = virsh.setmaxmem(vm_name, max_mem)
+        if mem_config == "without_numa":
+            libvirt.check_exit_status(ret)
+            libvirt_vmxml.check_guest_xml(vm_name, mem_xml % max_mem)
```
Comment (dzhengfy) [inline libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py:72]: same

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

File: libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py (hunk 7 above)
Comment (nanli1) [reply libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py]: done

File: libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py (hunk 5 above)
Comment (nanli1) [reply libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py]: done

File: libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py (hunk 4 above)
Comment (nanli1) [reply libvirt/tests/src/mem/memory_allocation/memory_allocation_config_modification.py]: done

File: libvirt/tests/cfg/mem/memory_allocation/memory_allocation_config_modification.cfg (hunk 2 above)
Comment (nanli1) [reply libvirt/tests/cfg/mem/memory_allocation/memory_allocation_config_modification.cfg]: done

**Review by chunfuwen — APPROVED** [review]
lgtm

**Review by dzhengfy — APPROVED** [review]

**Pull request merged**